### PR TITLE
[Intel MKL] BF16 enabling for MklLayoutPass test - v5

### PR DESCRIPTION
**Pull request by noim210**
[Intel MKL] BF16 enabling for MklLayoutPass test - v5
Signed-off-by: mazharul <mazharul.islam@intel.com>

### CODE REVIEW

**Comment by nhasabni** [comment]
@googlebot I consent.

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR and I'm so sorry for my long delay!

**Comment by gbaned** [comment]
@noim210 Can you please check @penpornk's comments and keep us posted ? Thanks!

**Comment by gbaned** [comment]
@noim210  Any update on this PR? Please. Thanks!

**Comment by noim210** [comment]
Yes, I am almost done with the requested review. I am going to push soon.

File: tensorflow/core/common_runtime/mkl_layout_pass_test.cc [hunk 1]
```diff
@@ -3464,428 +3503,485 @@ TEST_F(MklLayoutPassTest, NodeRewrite_QuantizedDepthwiseConv2D_Positive) {
...
+      DoMklLayoutOptimizationPass(),                                         \
+      "A(" #INPUT ");B(_MklLRN);C(_MklMaxPool);D(" #INPUT ");DMT/_0(Const);" \
+      "DMT/_1(Const);DMT/_2(Const);E(_MklMaxPoolGrad);F(" #INPUT ");"        \
+      "G(_MklLRNGrad);H(" #INPUT ");I(Zeta)|A->B;A:control->DMT/_0:control;" \
+      "B->C;B->E;B->G:2;B:1->G:3;B:2->C:1;B:2->E:4;B:2->G:6;B:3->G:7;"       \
+      "B:control->DMT/_1:control;C->E:1;C:1->E:3;C:2->E:5;C:3->E:7;D->E:2;"  \
+      "DMT/_0->B:1;DMT/_1->E:6;DMT/_2->G:5;E->G;E:1->G:4;"                   \
+      "E:control->DMT/_2:control;F->G:1;G->I:1;H->I");                       \
 }
+REGISTER_TEST_FLOAT32(MaxPoolLRN_Positive);
```
[the remark this answers is not in the document]
Comment (noim210) [reply tensorflow/core/common_runtime/mkl_layout_pass_test.cc:3613]: Yes, we might expand it to bfloat16, when bfloat16 will be available for this.

File: tensorflow/core/common_runtime/mkl_layout_pass_test.cc [hunk 2]
```diff
@@ -3064,276 +3062,318 @@ REGISTER_TEST_ALL_TYPES(NodeRewrite_TanhGrad_Positive);
...
-      " attr { key: 'epsilon'      value { f: 0.0001 } }"
-      " attr { key: 'is_training'  value { b: true } }"
-      " input: ['A', 'B', 'C', 'D', 'E'] }"
-      "node { name: 'G' op: 'Zeta' attr { key: 'T' value { type: DT_HALF } }"
-      " input: ['A', 'F'] }");
-  EXPECT_EQ(DoMklLayoutOptimizationPass(),
-            "A(HalfInput);B(HalfInput);C(Input);D(Input);E(Input);"
-            "F(FusedBatchNormGradV2);G(Zeta)|A->F;A->G;"
-            "B->F:1;C->F:2;D->F:3;E->F:4;F->G:1");
+#define REGISTER_TEST(NAME, T, INPUT)                                         \
```
[the remark this answers is not in the document]
Comment (noim210) [reply tensorflow/core/common_runtime/mkl_layout_pass_test.cc]: I updated the code according to your suggestion

File: tensorflow/core/common_runtime/mkl_layout_pass_test.cc [hunk 3]
```diff
@@ -3064,276 +3062,318 @@ REGISTER_TEST_ALL_TYPES(NodeRewrite_TanhGrad_Positive);
...
-      " attr { key: 'epsilon'      value { f: 0.0001 } }"
-      " attr { key: 'is_training'  value { b: true } }"
-      " input: ['A', 'B', 'C', 'D', 'E'] }"
-      "node { name: 'G' op: 'Zeta' attr { key: 'T' value { type: DT_HALF } }"
-      " input: ['A', 'F'] }");
-  EXPECT_EQ(DoMklLayoutOptimizationPass(),
-            "A(HalfInput);B(Input);C(Input);D(Input);E(Input);"
-            "F(FusedBatchNormV2);G(Zeta)|A->F;A->G;"
-            "B->F:1;C->F:2;D->F:3;E->F:4;F->G:1");
+#define REGISTER_TEST(NAME, T, INPUT)                                         \
```
[the remark this answers is not in the document]
Comment (noim210) [reply tensorflow/core/common_runtime/mkl_layout_pass_test.cc]: I updated the code according to your suggestion

File: tensorflow/core/common_runtime/mkl_layout_pass_test.cc [hunk 4]
```diff
@@ -3064,276 +3062,318 @@ REGISTER_TEST_ALL_TYPES(NodeRewrite_TanhGrad_Positive);
...
-      " attr { key: 'epsilon'      value { f: 0.0001 } }"
-      " attr { key: 'is_training'  value { b: true } }"
-      " input: ['A', 'B', 'C', 'D', 'E'] }"
-      "node { name: 'G' op: 'Zeta' attr { key: 'T' value { type: DT_HALF } }"
-      " input: ['A', 'F'] }");
-  EXPECT_EQ(DoMklLayoutOptimizationPass(),
-            "A(HalfInput);B(Input);C(Input);D(Input);E(Input);"
-            "F(FusedBatchNormV3);G(Zeta)|A->F;A->G;"
-            "B->F:1;C->F:2;D->F:3;E->F:4;F->G:1");
+#define REGISTER_TEST(NAME, T, INPUT)                                         \
```
[the remark this answers is not in the document]
Comment (noim210) [reply tensorflow/core/common_runtime/mkl_layout_pass_test.cc]: I updated the code according to your suggestion

File: tensorflow/core/common_runtime/mkl_layout_pass_test.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (penpornk) [reply tensorflow/core/common_runtime/mkl_layout_pass_test.cc:3604]: Thank you for the clarification!

**Review by penpornk — APPROVED** [review]
Thank you again for the PR!

**Pull request merged**